src/core/lender/assessor.sol

```diff
@@ -78,13 +78,13 @@ contract Assessor is Math, DSNote, Auth {
         return _calcSeniorAssetValue(poolValue, trancheReserve, SeniorTrancheLike(tranche).debt(), _juniorReserve());
     }
 
-    function calcTokenPrice() public returns (uint) {
-        return mul(_calcTokenPrice(), tokenAmountForONE);
+    function calcTokenPrice(address tranche) public returns (uint) {
```

Comment:
The method is called by the operator and not the tranche.

That's the reason why `msg.sender` is not working.